charts/spire/values.yaml

```diff
@@ -1,31 +1,45 @@
 ---
 # Default values for spire.
 # This is a YAML-formatted file.
-# Declare variables to be passed into your templates.
-
-# workaround since Azure doesn't support by default 10250 security port for kubelet
-azure: false
-# aws - enables the plugin for AWS NodeAttestor
-aws: false
-# openShift requires special configuration, including different security level
-openShift: false
 
 # namespace - namespace for deploying SPIRE agents and workload registrar
 namespace: spire
-
-# SPIRE specific values
-# clustername needs to match between spire-config, k8s-workload-registrar config, and the actual cluster name
+# clustername needs to match between agent-config, k8s-workload-registrar config,
+# and the actual cluster name
 clustername: spire-example
 # region - arbitrary label to describe the region
 region: sample-region
-# trustdomain is arbitrary but needs to match between spire-config and k8s-workload-registrar config
-# for multi-cluster support, trustdomain must be identical for all clusters and
+
+# SPIRE related elements
+# trustdomain is arbitrary but needs to match between agent-config and
+# k8s-workload-registrar config and SPIRE Server.
+# For multi-cluster support, trustdomain must be identical for all clusters and
 # SPIRE server
 trustdomain: spiretest.com
-
-# SPIRE workload registrar values:
-spireAddress: spire-server
-spirePort: 443
-spireRegistrarImg: gcr.io/spiffe-io/k8s-workload-registrar
-spireAgentImg: gcr.io/spiffe-io/spire-agent
+# SPIRE version:
 spireVersion: 1.0.2
+
+# spireServer - location of the SPIRE server
+spireServer:
+  address: spire-server.appdomain.cloud
+  port: 443
+
+# SPIRE Agent related configuration
+spireAgent:
+  img: gcr.io/spiffe-io/spire-agent
+  # SPIRE Agent socket:
+  socketDir: /run/spire/sockets
+  socketFile: agent.sock
+
+# Workload Registrar configuration
+spireRegistrar:
+  img: gcr.io/spiffe-io/k8s-workload-registrar
```

Comment:
probably would be nice here to be able to specify the identity template and label configuration here (using our defaults)

Reply:
This is a nice suggestion, but it is a bit challenging if we want to include `context` values that are dynamic and depend on the other helm variables. This should be investigated further, so I created a new issue #115